File: core/src/main/java/eu/kotkas/villak/core/model/Category.java
package eu.kotkas.villak.core.model;

import eu.kotkas.villak.core.model.enums.NameState;
import eu.kotkas.villak.core.util.IdUtil;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.List;

/**
 * @author Kristen Kotkas
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Category implements Serializable {

  @Builder.Default
  private int id = IdUtil.getId();
  private String name;
  private List<Question> questions;
  @Builder.Default
  private NameState nameState = NameState.CATEGORY_HIDE;

}

File: web/src/app/modules/villak/game-data/model/category.ts
import {Action} from './action';
import {Question} from './question';

export class Category {
  id: number;
  name: string;
  questions: Question[];
  nameState: Action.CATEGORY_SHOW | Action.CATEGORY_HIDE = Action.CATEGORY_HIDE;

  constructor(id: number, name: string, questions: Question[]) {
    this.id = id;
    this.name = name;
    this.questions = questions;
  }
}
